torch_geometric/nn/dense/linear.py

```diff
@@ -161,42 +163,46 @@ def __repr__(self) -> str:
 class HeteroLinear(torch.nn.Module):
     r"""Applies separate linear tranformations to the incoming data according
     to types
-
     .. math::
         \mathbf{x}^{\prime}_{\kappa} = \mathbf{x}_{\kappa}
         \mathbf{W}^{\top}_{\kappa} + \mathbf{b}_{\kappa}
-
     for type :math:`\kappa`.
     It supports lazy initialization and customizable weight and bias
     initialization.
-
     Args:
         in_channels (int): Size of each input sample. Will be initialized
             lazily in case it is given as :obj:`-1`.
         out_channels (int): Size of each output sample.
         num_types (int): The number of types.
         **kwargs (optional): Additional arguments of
             :class:`torch_geometric.nn.Linear`.
-
     Shapes:
         - **input:**
           features :math:`(*, F_{in})`,
           type vector :math:`(*)`
         - **output:** features :math:`(*, F_{out})`
     """
     def __init__(self, in_channels: int, out_channels: int, num_types: int,
-                 **kwargs):
+                 is_sorted: Optional[bool] = False, **kwargs):
         super().__init__()
 
         self.in_channels = in_channels
         self.out_channels = out_channels
-
+        self._WITH_PYG_LIB = torch.cuda.is_available() and _WITH_PYG_LIB
+        self.is_sorted = is_sorted
         self.lins = torch.nn.ModuleList([
             Linear(in_channels, out_channels, **kwargs)
             for _ in range(num_types)
         ])
 
         self.reset_parameters()
+        if self._WITH_PYG_LIB:
```

Comment:
Can't we instead initialize a `Parameter` with shape `[num_types, in_channels, out_channels]`?

Reply:
well i figured this way we preserve the kwargs and initiliazers of linear without having to re-code that specifically for the pyg_lib case